### PR TITLE
Update rubocop.yml

### PR DESCRIPTION
**Pull request by damiendillon**
Update outdated `rubocop.yml` to satisfy complaints.

Do not change any settings. Add new categories of
-Security
-Layout
-Naming
and move appropriate cops here.

Rename cops as required and delete those that are redundant.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 AllCops:
   DisabledByDefault: true
-  TargetRubyVersion: 2.3
```
Comment (dsager) [inline .rubocop.yml]: In theory we support 2.3 upwards, see [here](https://github.com/Devex/article_json/blob/master/article_json.gemspec#L26). So I'm not sure if RuboCop should target the latest version. My inclination would be to target the minimal required version.
But we can also increase the minimum required ruby version and drop support for older versions ([Actually all versions <3 have already reached EOL](https://www.ruby-lang.org/en/downloads/branches/)).

Comment (damiendillon) [reply .rubocop.yml]: Hmm. I can just revert this to 2.3 and if we 
>increase the minimum required ruby version and drop support for older versions ([Actually all versions <3 have already reached EOL](https://www.ruby-lang.org/en/downloads/branches/)).

that can be a separate task.

Comment (damiendillon) [reply .rubocop.yml]: Reverted to 2.3

**Comment by ozgun** [comment]
:+1:

**Pull request merged**